src/TestUtilities/TestingMessageConsumers/TestedMessageConsumer.php

```diff
@@ -23,10 +25,16 @@ public function handle(Message $message): void
         if (++$this->numberOfMessagesProcessed === $this->failAfterNumberOfMessages) {
             throw new LogicException('Too many messages');
         }
+        $this->lastProcessedUuid = $message->aggregateRootId();
     }
 
     public function numberOfMessagesProcessed(): int
     {
         return $this->numberOfMessagesProcessed;
     }
+
+    public function lastProcessedUuid(): ?AggregateRootId
+    {
+        return $this->lastProcessedUuid ?? null;
```

Comment:
Isn't ?? null redundant?

Reply:
without ?? it would throw a  `Typed property EventSauce\EventSourcing\TestUtilities\TestingMessageConsumers\TestedMessageConsumer::$lastProcessedUuid must not be accessed before initialization` when the method is called before a message is handled.

Reply:
I think with a null default it can be removed. I'll check it after the merge :)

Reply:
Yeah, should work 👍